src/backend/pluginPlan.test.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+
+import {Graph} from "../core/graph";
+import {node as graphNode} from "../core/graphTestUtil";
+import {createProject} from "../core/project";
+import {TestTaskReporter} from "../util/taskReporter";
+import {pluginMirrorPlan} from "./pluginPlan";
+
+const githubSentinel = graphNode("github-sentinel");
+const githubGraph = () => new Graph().addNode(githubSentinel);
+const discourseSentinel = graphNode("discourse-sentinel");
+const discourseGraph = () => new Graph().addNode(discourseSentinel);
+const combinedGraph = () => Graph.merge([githubGraph(), discourseGraph()]);
+
+const mockCacheProvider = () => ({
+  database: jest.fn(),
+});
+
+const mockPluginLoaders = () => ({
+  github: {
+    updateMirror: jest.fn(),
+    createGraph: jest.fn().mockResolvedValue(githubGraph()),
+  },
+  discourse: {
+    updateMirror: jest.fn(),
+    createGraph: jest.fn().mockResolvedValue(discourseGraph()),
+  },
+  identity: {
+    contractGraph: jest.fn().mockResolvedValue(combinedGraph()),
+  },
+});
+
+describe("src/backend/pluginPlan", () => {
+  describe("pluginMirrorPlan", () => {
+    it("should update discourse mirror", async () => {
+      // Given
+      const loaders = mockPluginLoaders();
+      const cache = mockCacheProvider();
+      const reporter = new TestTaskReporter();
+      const ghToken = null;
+      const project = createProject({
+        id: "has-discourse",
+        discourseServer: {serverUrl: "http://foo.bar"},
+      });
+
+      // When
+      const mirror = pluginMirrorPlan(loaders, ghToken, cache, reporter);
+      await mirror(project);
+
+      // Then
+      const {discourse, github, identity} = loaders;
+      expect(discourse.updateMirror).toBeCalledTimes(1);
+      expect(discourse.updateMirror).toBeCalledWith(
+        project.discourseServer,
+        cache,
+        reporter
+      );
+      expect(discourse.createGraph).toBeCalledTimes(0);
+      expect(github.updateMirror).toBeCalledTimes(0);
+      expect(github.createGraph).toBeCalledTimes(0);
+      expect(identity.contractGraph).toBeCalledTimes(0);
+    });
+
+    it("should fail when missing GithubToken", async () => {
```

Comment:
It should fail when missing GH token, *and* repoIds are present in the project. If there are no repoIds, then missing GH token should not fail. Please rename this test case, and add another one in the case with no GH token and no repoIds.